### DISCUSSION TITLE
Is specifying ryuk from classpath supported ?

### DISCUSSION THREAD

**Discussion by rajjaiswalsaumya** [testcontainers/testcontainers-java#9344] [Q&A]
I am trying to upgrade to the latest spring boot, and we are upgrading test containers. Unfortunately, the latest test containers requirs a recent version of the Ryuk container. Since we don't allow docker to be downloaded from outside organizations, I created a clone of the docker image of Ryuk in the docker hub. Now, I am trying to specify the image name from the local testcontainer.properties file in the project test classpath since others require an old version of the Ryuk image. 

Unfortunately, this does not work. The configuration is imported via 
`@TestPropertySource(value = {"classpath:/testcontainers.properties"})`

Contents of src/test/resources/testcontainer.properties 

```
dockerconfig.source=autoIgnoringUserProperties
ryuk.container.image=<local-docker-repo>/testcontainers/ryuk:0.7.0
ryuk.wait.timeout=60000
testcontainers.wait.timeout=300000  # 300 seconds (5 minutes)
log.level=DEBUG

```

Contents of ~/.testcontainer.properties 

```
docker.client.strategy=org.testcontainers.dockerclient.UnixSocketClientProviderStrategy
ryuk.container.image=<local-docker-repo>/testcontainers/ryuk:0.2.3
```

I want to download `0.7.0` version in my project test and not `0.2.3`. What is the best way I can achieve this?

I can see and feel the code for [TestcontainersConfiguration.java#getConfigurable](https://github.com/testcontainers/testcontainers-java/blob/main/core/src/main/java/org/testcontainers/utility/TestcontainersConfiguration.java#L225) requires slight modification. It is currently not overriding the property defined late in the order. 

E.g. if a classpath properties is present and the same is in user home, it is using user home property and not the one present in classpath. This is weird behavior. Ideally override should happen in given order by testcontainers.

**Comment by eddumelendez** [comment]
Hi, the name should be `src/test/resources/testcontainers.properties`, see at the missing `s`. That approach should work. Look also at the different approaches to use [Image Name Substitution](https://java.testcontainers.org/features/image_name_substitution/)
